zerver/views/streams.py

```diff
@@ -113,6 +113,16 @@ def list_subscriptions_backend(request, user_profile):
     # type: (HttpRequest, UserProfile) -> HttpResponse
     return json_success({"subscriptions": gather_subscriptions(user_profile)[0]})
 
+@has_request_variables
+def update_subscriptions_property(request, user_profile, stream_name=REQ(), property=REQ(), value=REQ()):
+    # type: (HttpRequest, UserProfile, str, str, str) -> HttpResponse
+    (stream, recipient, sub) = access_stream_by_name(user_profile, stream_name)
+    if sub is None:
+        return json_error(_("Not subscribed to stream %s") % (stream_name,))
+    do_change_subscription_property(user_profile, sub, stream,
+                                    property, ujson.loads(value))
+    return json_success()
```

Comment:
Rather than adding a new endpoint, why not just modify the existing `json_subscription_property` endpoint?

Also, we've been trying to migrate the API to access streams by ID, since that's more robust; probably worth doing this right as part of the migration.  

So I think for the "mute streams" piece, we should do a few commits
(1) Migration the existing /json endpoint to work on IDs, not names
(2) Move that endpoint to be API-accessible